Core/src/game/functions/ints/count/sizeBiggestGroup/CountSizeBiggestGroup.java

```diff
@@ -0,0 +1,348 @@
+ * @author Eric.Piette & Cedric.Antoine
+ */
+public final class CountSizeBiggestGroup extends BaseIntFunction
+{
+	private static final long serialVersionUID = 1L;
+
+	//-------------------------------------------------------------------------
+
+	/** The graph element type. */
+	private SiteType type;
+
+	/** The condition */
+	private final BooleanFunction condition;
+	
+	/** The visibility condition */
+	private final BooleanFunction isVisibleFn;
+
+	/** Direction chosen. */
+	private final DirectionsFunction dirnChoice;
+
+	//-------------------------------------------------------------------------
+
+	/**
+	 * @param type       The graph element type [default SiteType of the board].
+	 * @param directions The directions of the connection between elements in the
+	 *                   group [Adjacent].
+	 * @param If         The condition on the pieces to include in the group.
+	 * @param isVisible  Visibility condition for group pieces
+	 */
+	public CountSizeBiggestGroup
+	(
+		@Opt 	        final SiteType        type,
+		@Opt            final Direction       directions,
+		@Opt @Or @Name  final BooleanFunction If,
+		@Opt @Or @Name  final BooleanFunction isVisible
+	)
+	{
+		this.type = type;
+		dirnChoice = (directions != null) ? directions.directionsFunctions()
+				: new Directions(AbsoluteDirection.Adjacent, null);
+		condition = (If != null) ? If : new IsOccupied(type, To.construct());
+		isVisibleFn = isVisible;
+	}
+
+	//-------------------------------------------------------------------------
+
+	@Override
+	public int eval(final Context context)
+	{
+		final Topology topology = context.topology();
+		final List<? extends TopologyElement> sites = context.topology().getGraphElements(type);
+		final ContainerState cs = context.containerState(0);
+		final int origTo = context.to();
+
+		int count = 0;
+
+		final BitSet sitesChecked = new BitSet(sites.size());
+		final TIntArrayList sitesToCheck = new TIntArrayList();
+
+		if (context.game().isDeductionPuzzle())
+		{
+			for (int site = 0; site < sites.size(); site++)
+				if (cs.what(site, type) != 0)
+					sitesToCheck.add(site);
+		}
+		else
+		{
+			for(final TopologyElement element: sites)
+			{
+				context.setTo(element.index());
+				if(condition.eval(context))
+					sitesToCheck.add(element.index());
+			}
+		}
+
+		for (int k = 0; k < sitesToCheck.size(); k++)
+		{
+			final int from = sitesToCheck.getQuick(k);
+			
+			if (sitesChecked.get(from))
+				continue;
+			
+			// Good to use both list and BitSet here at the same time for different advantages
+			final TIntArrayList groupSites = new TIntArrayList();
+			final BitSet groupSitesBS = new BitSet(sites.size());
+			
+			context.setTo(from);
+			
+			if (isVisibleFn != null && isVisibleFn.eval(context) == true && ((from == 8 && cs.what(9, type) != 0) || (from == 10 && cs.what(11, type) != 0) || (from == 14 && cs.what(15, type) != 0) || (from == 18 && cs.what(19, type) != 0) || (from == 20 && cs.what(21, type) != 0)))
```

Comment:
magic numbers (need commenting and/or other safety checks as discussed above for one of the other files)